### PR TITLE
✨ Add new cheatcodes to Hevm.sol

### PR DESCRIPTION
**Pull request by Joeysantoro**
## Description

Update Hevm to include the latest cheatcodes and comments from https://github.com/gakonst/foundry/tree/master/forge.

I know @transmissions11 didn't like the comments last time this was proposed because of solmate coding style. I think an exception can be made here for users to have a reference directly in their IDE to the cheatcodes and their functionality.

resolves https://github.com/Rari-Capital/solmate/issues/168

## Checklist

Not needed bc interface change only

Ensure you completed **all of the steps** below before submitting your pull request:

- [ ] Ran `dapp snapshot`?
- [ ] Ran `npm run lint`?
- [ ] Ran `forge test`?
- [ ] Ran `dapp test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

**Comment by transmissions11** [comment]
> I know @transmissions11 didn't like the comments last time this was proposed because of solmate coding style. I think an exception can be made here for users to have a reference directly in their IDE to the cheatcodes and their functionality.

ya fair, but shouldn't we use @notice so it shows up when you hover over them?

**Comment by Joeysantoro** [comment]
> ya fair, but shouldn't we use @notice so it shows up when you hover over them?

Done!

File: src/test/utils/Hevm.sol [hunk 1]
```diff
@@ -1,61 +1,61 @@
...
     function record() external;
-    // Gets all accessed reads and write slot from a recording session, for a given address
+    /// @notice Gets all accessed reads and write slot from a recording session, for a given address
     function accesses(address) external returns (bytes32[] memory reads, bytes32[] memory writes);
-    // Prepare an expected log with (bool checkTopic1, bool checkTopic2, bool checkTopic3, bool checkData).
-    // Call this function, then emit an event, then call a function. Internally after the call, we check if
-    // logs were emitted in the expected order with the expected topics and data (as specified by the booleans)
+    /// @notice Prepare an expected log with (bool checkTopic1, bool checkTopic2, bool checkTopic3, bool checkData).
+    /// @notice Call this function, then emit an event, then call a function. Internally after the call, we check if
+    /// @notice logs were emitted in the expected order with the expected topics and data (as specified by the booleans)
```
Comment (transmissions11) [inline src/test/utils/Hevm.sol:43]:
```suggestion
    /// logs were emitted in the expected order with the expected topics and data (as specified by the booleans)
```

File: src/test/utils/Hevm.sol [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
     function expectEmit(bool,bool,bool,bool) external;
-    // Mocks a call to an address, returning specified data.
-    // Calldata can either be strict or a partial match, e.g. if you only
-    // pass a Solidity selector to the expected calldata, then the entire Solidity
-    // function will be mocked.
+    /// @notice Mocks a call to an address, returning specified data.
+    /// @notice Calldata can either be strict or a partial match, e.g. if you only
+    /// @notice pass a Solidity selector to the expected calldata, then the entire Solidity
+    /// @notice function will be mocked.
```
Comment (transmissions11) [inline src/test/utils/Hevm.sol:48]:
```suggestion
    /// pass a Solidity selector to the expected calldata, then the entire Solidity
    /// function will be mocked.
```

**Pull request merged**